### PR TITLE
add warm start attribute to HpBandSterSearchCV, so that the user can choose whether to warm start

### PR DESCRIPTION
**Pull request by poetair**
https://github.com/Yard1/hpbandster-sklearn/issues/6
add warm start attribute to HpBandSterSearchCV, so that the user can choose whether to warm start

**warm_start = True(default)**

code
---

import numpy as np
from sklearn.datasets import load_iris
from sklearn.ensemble import RandomForestClassifier
from hpbandster_sklearn import HpBandSterSearchCV
from sklearn.model_selection import KFold, cross_val_score

import ConfigSpace as CS
import ConfigSpace.hyperparameters as CSH

X, y = load_iris(return_X_y=True)
clf = RandomForestClassifier(n_estimators=10, random_state=0)  
kf = KFold(shuffle=False)

param_distributions = CS.ConfigurationSpace(seed=0)
param_distributions.add_hyperparameter(CSH.UniformIntegerHyperparameter("min_samples_split", 2, 11))
param_distributions.add_hyperparameter(CSH.UniformIntegerHyperparameter("max_depth", 2, 4))

search = HpBandSterSearchCV(clf, param_distributions, resource_name='n_estimators', min_budget=10, max_budget=80,
                            random_state=0, n_jobs=1,cv=kf, warm_start=True,
                            n_iter=4, **{'eta':2}).fit(X, y)

for p, s, c in zip(search.cv_results_['params'], search.cv_results_['n_resources'], search.cv_results_['mean_test_score']):
    params = p
    params['n_estimators'] = s
    clf = clf.set_params(**params)
    
    print(params)
    print(np.mean(cross_val_score(clf , X, y, cv=kf)), c)

output：Some are different due to warm start 
---
{'max_depth': 3, 'min_samples_split': 9, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 3, 'min_samples_split': 9, 'n_estimators': 20}
0.9133333333333333 0.9133333333333333
{'max_depth': 3, 'min_samples_split': 9, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 9, 'n_estimators': 80}
0.9 0.9
{'max_depth': 3, 'min_samples_split': 7, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 3, 'min_samples_split': 7, 'n_estimators': 20}
0.9133333333333333 0.9133333333333333
{'max_depth': 3, 'min_samples_split': 7, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 20}
0.9133333333333333 0.9133333333333333
{'max_depth': 3, 'min_samples_split': 10, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 3, 'min_samples_split': 10, 'n_estimators': 20}
0.9133333333333333 0.9133333333333333
{'max_depth': 4, 'min_samples_split': 5, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 4, 'min_samples_split': 7, 'n_estimators': 10}
0.9133333333333333 0.9199999999999999
{'max_depth': 3, 'min_samples_split': 7, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 3, 'min_samples_split': 11, 'n_estimators': 20}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 11, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 11, 'n_estimators': 80}
0.9 0.9
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 20}
**0.9133333333333333 0.9066666666666666**(inequality)
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 6, 'n_estimators': 20}
**0.9133333333333333 0.9066666666666666**(inequality)
{'max_depth': 2, 'min_samples_split': 2, 'n_estimators': 20}
**0.9133333333333333 0.9066666666666666**(inequality)
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 80}
0.9 0.9
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 80}
0.9 0.9
{'max_depth': 3, 'min_samples_split': 10, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 2, 'n_estimators': 40}
**0.9 0.9066666666666666**(inequality)
{'max_depth': 2, 'min_samples_split': 10, 'n_estimators': 80}
**0.9066666666666666 0.9**(inequality)
{'max_depth': 3, 'min_samples_split': 11, 'n_estimators': 80}
0.9 0.9
{'max_depth': 3, 'min_samples_split': 9, 'n_estimators': 80}
0.9 0.9
{'max_depth': 4, 'min_samples_split': 10, 'n_estimators': 80}
0.9 0.9

**warm_start = False**

code
---
import numpy as np
from sklearn.datasets import load_iris
from sklearn.ensemble import RandomForestClassifier
from hpbandster_sklearn import HpBandSterSearchCV
from sklearn.model_selection import KFold, cross_val_score

import ConfigSpace as CS
import ConfigSpace.hyperparameters as CSH

X, y = load_iris(return_X_y=True)
clf = RandomForestClassifier(n_estimators=10, random_state=0)  
kf = KFold(shuffle=False)

param_distributions = CS.ConfigurationSpace(seed=0)
param_distributions.add_hyperparameter(CSH.UniformIntegerHyperparameter("min_samples_split", 2, 11))
param_distributions.add_hyperparameter(CSH.UniformIntegerHyperparameter("max_depth", 2, 4))

search = HpBandSterSearchCV(clf, param_distributions, resource_name='n_estimators', min_budget=10, max_budget=80,
                            random_state=0, n_jobs=1,cv=kf, warm_start=False,
                            n_iter=4, **{'eta':2}).fit(X, y)

for p, s, c in zip(search.cv_results_['params'], search.cv_results_['n_resources'], search.cv_results_['mean_test_score']):
    params = p
    params['n_estimators'] = s
    clf = clf.set_params(**params)
    
    print(params)
    print(np.mean(cross_val_score(clf , X, y, cv=kf)), c)

output：All the same
---
{'max_depth': 3, 'min_samples_split': 9, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 3, 'min_samples_split': 9, 'n_estimators': 20}
0.9133333333333333 0.9133333333333333
{'max_depth': 3, 'min_samples_split': 9, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 9, 'n_estimators': 80}
0.9 0.9
{'max_depth': 3, 'min_samples_split': 7, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 3, 'min_samples_split': 7, 'n_estimators': 20}
0.9133333333333333 0.9133333333333333
{'max_depth': 3, 'min_samples_split': 7, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 20}
0.9133333333333333 0.9133333333333333
{'max_depth': 3, 'min_samples_split': 10, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 3, 'min_samples_split': 10, 'n_estimators': 20}
0.9133333333333333 0.9133333333333333
{'max_depth': 4, 'min_samples_split': 5, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 4, 'min_samples_split': 7, 'n_estimators': 10}
0.9133333333333333 0.9133333333333333
{'max_depth': 3, 'min_samples_split': 7, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 10}
0.9199999999999999 0.9199999999999999
{'max_depth': 3, 'min_samples_split': 11, 'n_estimators': 20}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 20}
0.9133333333333333 0.9133333333333333
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 80}
0.9 0.9
{'max_depth': 3, 'min_samples_split': 6, 'n_estimators': 20}
0.9133333333333333 0.9133333333333333
{'max_depth': 3, 'min_samples_split': 6, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 2, 'min_samples_split': 2, 'n_estimators': 20}
0.9133333333333333 0.9133333333333333
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 80}
0.9 0.9
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 80}
0.9 0.9
{'max_depth': 3, 'min_samples_split': 11, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 9, 'n_estimators': 40}
0.9066666666666666 0.9066666666666666
{'max_depth': 2, 'min_samples_split': 10, 'n_estimators': 80}
0.9066666666666666 0.9066666666666666
{'max_depth': 3, 'min_samples_split': 7, 'n_estimators': 80}
0.9 0.9
{'max_depth': 3, 'min_samples_split': 8, 'n_estimators': 80}
0.9 0.9
{'max_depth': 4, 'min_samples_split': 10, 'n_estimators': 80}
0.9 0.9

### CODE REVIEW

**Review by Yard1 — APPROVED** [review]
Thanks, this looks good.

File: hpbandster_sklearn/HpBandSterSearchCV.py [hunk 1]
```diff
@@ -496,6 +503,7 @@ def fit(self, X, y, groups=None, **fit_params):
                     cv_n_splits=n_splits,
                     groups=groups,
                     scoring=scorers,
+                    warm_start = self.warm_start
```
Comment (Yard1) [inline hpbandster_sklearn/HpBandSterSearchCV.py]:
```suggestion
                    warm_start=self.warm_start,
```

**Comment by poetair** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        
            Thanks, and could you release a new version? because some of your updates are not included in 2.0.1.
        
    
    
        
            
        
    
    
        
        

        
    

    On 10/11/2022 18:44，Antoni ***@***.***> wrote： 



@Yard1 approved this pull request.

Thanks, this looks good.

**Comment by Yard1** [comment]
Yes, I'll release a new version today or tomorrow!

**Comment by poetair** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        Thanks, and here I wrote two issues. Could you check if there is such a problem in the code?https://github.com/Yard1/hpbandster-sklearn/issues/5 https://github.com/Yard1/hpbandster-sklearn/issues/4 
        
    
    
        
            
        
    
    
        
        

        
    

    On 10/11/2022 21:32，Antoni ***@***.***> wrote： 



Yes, I'll release a new version today or tomorrow!

**Comment by poetair** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        
            Thanks for your reply, and  I put forward the following suggestions. Could you reflect these suggestions in the latest version, if you think it is reasonable, thanks.
    
        
            https://github.com/Yard1/hpbandster-sklearn/issues 
        
    https://github.com/Yard1/hpbandster-sklearn/pulls 
    
        
        

        
    

    On 10/11/2022 21:32，Antoni ***@***.***> wrote： 



Yes, I'll release a new version today or tomorrow!
